app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -82,18 +82,20 @@ def feeds
     end
 
     def machine_id(feed)
-      os_resource_for(feed).properties['Machine Id']
+      os_resource_for(feed).try(:properties).try { |prop| prop['Machine Id'] }
     end
 
     def os_resource_for(feed)
       with_provider_connection do |connection|
         os = os_for(feed)
-        os_resources = connection.inventory.list_resources_for_type(os.path, true)
-        unless os_resources.nil? || os_resources.empty?
-          return os_resources.first
+        unless os.nil?
+          os_resources = connection.inventory.list_resources_for_type(os.path, true)
+          unless os_resources.nil? || os_resources.empty?
+            return os_resources.first
+          end
         end
 
-        $mw_log.warn "Found no OS resources for resource type #{os.path}"
+        $mw_log.warn "Found no OS resources for resource type #{os.try(:path)}"
```

Comment:
You already have a conditional for `os.nil?`, would it make sense to move this into that block and have a different log message if os is nil?  Or maybe change it to print out `#{feed}` since `Found no OS resources for resource type nil` isn't going to be very helpful for debugging?